src/cadenza.js

```diff
@@ -831,6 +844,10 @@ export class CadenzaClient {
   }
 
   #createUrl(/** @type string */ path, /** @type URLSearchParams */ params) {
+    assert(
+      !this.#isEmbeddedInCadenza,
```

Comment:
This method is called also from fetchData and downloadData where you don't need an embedded Cadenza, so the name of this getter is confusing.

Proposal: Inline this.#baseUrl != null here.